src/primitives/transaction.h

```diff
@@ -224,10 +224,13 @@ class CTransaction
     void UpdateHash() const;
 
 public:
-    static const int32_t STANDARD_VERSION = 1;
-    static const int32_t SAPLING_VERSION = 2;
+    enum TxVersion {
+        LEGACY      = 1,
+        SAPLING     = 2,
+        TOOHIGH
```

Comment:
are we totally sure that this will behave correctly in every system architecture?
wouldn't be better to define a value=3 for `TOOHIGH` as well?

Reply:
Yes, this is the intended behavior.

> For any other enumerator whose definition does not have an initializer, the associated value is the value of the previous enumerator plus one.
```
enum Foo { a, b, c = 10, d, e = 1, f, g = f + c };
//a = 0, b = 1, c = 10, d = 11, e = 1, f = 2, g = 12
```
<https://en.cppreference.com/w/cpp/language/enum>

The reason why I didn't explicitely set it to `3` is just do not have to modify it, if we add another version.
We basically use the same trick, e.g., with the `UpgradeIndex` enum (checking against `MAX_NETWORK_UPGRADES`)

Reply:
ok great, then all good.